packages/clerk-js/src/ui/elements/Navbar.tsx

```diff
@@ -69,40 +71,56 @@ export const NavBar = (props: NavBarProps) => {
     }
   };
 
-  useSafeLayoutEffect(function selectNavItemBasedOnVisibleSection() {
-    const mountObservers = () => {
-      const ids = routes.map(r => r.id);
-      const sectionElements = ids
-        .map(getSectionId)
-        .map(id => contentRef.current?.querySelector(id))
-        .filter(s => s);
-
-      if (sectionElements.length === 0) {
-        return false;
-      }
-
+  useSafeLayoutEffect(
+    function selectNavItemBasedOnVisibleSection() {
       const callback: IntersectionObserverCallback = entries => {
         for (const entry of entries) {
           const id = entry.target?.id?.split('section-')[1];
           if (entry.isIntersecting && id) {
-            return setActiveId(id as unknown as any);
+            return setActiveId(id);
           }
         }
       };
-
       const observer = new IntersectionObserver(callback, { root: contentRef.current, threshold: 1 });
-      sectionElements.forEach(section => section && observer.observe(section));
-      return true;
-    };
 
-    const intervalId = setInterval(() => {
-      if (mountObservers()) {
-        clearInterval(intervalId);
-      }
-    }, 50);
+      const mountObservers = () => {
+        const ids = routes.map(r => r.id);
+        const sectionElements = ids
+          .map(getSectionId)
+          .map(id => contentRef.current?.querySelector(id))
+          .filter(s => s);
+
+        if (sectionElements.length === 0) {
+          return false;
+        }
 
-    return () => clearInterval(intervalId);
-  }, []);
+        sectionElements.forEach(section => section && observer.observe(section));
+        return true;
+      };
+
+      const intervalId = setInterval(() => {
+        if (mountObservers()) {
+          clearInterval(intervalId);
+        }
+      }, 50);
+
+      return () => {
+        observer.disconnect();
+      };
+    },
+    [router],
+  );
+
+  useEffect(() => {
+    routes.every(route => {
+      const isRoot = router.currentPath === router.fullPath && route.path === '/';
```

Comment:
Lets say we do this. 
```suggestion
      const isRoot = route.path === '/';
```
Which scenarios break the above?

Reply:
We are iterating through the navbar routes, so the `route.path` with be `'/'` at some point but we also need to make sure the current path is also the root. If we don't make both checks, the active id will be set as the root at all times, even if the `router.path` is not on the root.

Reply:
Ah I see. I misread `route` as `router` and the above snippet didn't make sense; sorry about that.

This is not related to this PR, however, it's not clearly apparent what the difference between these two is `router.currentPath === router.fullPath`. Noted so we can discuss later. Thanks!